File: bannerlayout/src/main/java/com/wkz/bannerlayout/utils/BannerSelectorUtils.java
package com.wkz.bannerlayout.utils;

import android.content.Context;
import android.graphics.drawable.GradientDrawable;
import android.graphics.drawable.StateListDrawable;
import android.support.annotation.NonNull;

public class BannerSelectorUtils {

    @NonNull
    public static StateListDrawable getDrawableSelector(Context context, float enabledRadius, int enabledColor, float normalRadius, int normalColor) {
        StateListDrawable drawable = new StateListDrawable();
        drawable.addState(new int[]{android.R.attr.state_enabled}, getShape(context, enabledRadius, enabledColor));
        drawable.addState(new int[]{-android.R.attr.state_enabled}, getShape(context, normalRadius, normalColor));
        return drawable;
    }

    @NonNull
    public static GradientDrawable getShape(Context context, float radius, int color) {
        GradientDrawable gd = new GradientDrawable();
        gd.setShape(GradientDrawable.RECTANGLE);
        gd.setCornerRadius(DisplayUtils.dp2px(context, radius));
        gd.setColor(color);
        return gd;
    }
}

File: bannerlayout/src/main/java/com/fphoenixcorneae/bannerlayout/utils/BannerSelectorUtils.kt
package com.fphoenixcorneae.bannerlayout.utils

import android.R
import android.content.Context
import android.graphics.drawable.GradientDrawable
import android.graphics.drawable.StateListDrawable

object BannerSelectorUtils {
    fun getDrawableSelector(context: Context, enabledRadius: Float, enabledColor: Int, normalRadius: Float, normalColor: Int): StateListDrawable {
        val drawable = StateListDrawable()
        drawable.addState(intArrayOf(R.attr.state_enabled), getShape(context, enabledRadius, enabledColor))
        drawable.addState(intArrayOf(-R.attr.state_enabled), getShape(context, normalRadius, normalColor))
        return drawable
    }

    fun getShape(context: Context, radius: Float, color: Int): GradientDrawable {
        val gd = GradientDrawable()
        gd.shape = GradientDrawable.RECTANGLE
        gd.cornerRadius = DisplayUtils.dp2px(context, radius).toFloat()
        gd.setColor(color)
        return gd
    }
}